src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonProxy.cs

```diff
@@ -99,9 +99,7 @@ public ClusterSingletonProxy(string singletonManagerPath, ClusterSingletonProxyS
             _singletonPath = (singletonManagerPath + "/" + settings.SingletonName).Split('/');
             _identityId = CreateIdentifyId(_identityCounter);
 
-            _memberAgeComparer = settings.ConsiderAppVersion
-                ? MemberAgeOrdering.DescendingWithAppVersion
-                : MemberAgeOrdering.Descending;
+            _memberAgeComparer = Member.AgeOrdering;
```

Comment:
This is the bug fix - descending order of members by age means we always get THE YOUNGEST at the front of the `membersByAge` list. Members MUST BE IN ASCENDING ORDER, OLDEST FIRST. I'm not sure how far back this bug goes - at least to v1.5. I'll add another comment after I've dug that up.